test/gtest/ucs/test_memtype_cache.cc

```diff
@@ -41,32 +59,22 @@ class test_memtype_cache : public ucs::test_with_param<ucs_memory_type_t> {
         ucs_memory_type_t mem_type;
         ucs_status_t status = ucs_memtype_cache_lookup(m_memtype_cache, ptr,
                                                        size, &mem_type);
-        EXPECT_UCS_OK(status);
-        EXPECT_EQ(expected_type, mem_type) << "ptr=" << ptr << " size=" << size;
+        check_lookup(ptr, size, status, mem_type, true, expected_type);
     }
 
-    void test_lookup_notfound(void *ptr, size_t size) const {
+    void test_lookup_notfound(const void *ptr, size_t size) const {
         if (!size) {
             return;
         }
 
         ucs_memory_type_t mem_type;
         ucs_status_t status = ucs_memtype_cache_lookup(m_memtype_cache, ptr,
                                                        size, &mem_type);
-        /* memory type should be not-found or unknown */
-        EXPECT_TRUE((status == UCS_ERR_NO_ELEM) ||
-                    ((status == UCS_OK) && (mem_type == UCS_MEMORY_TYPE_LAST)))
-              << "ptr=" << ptr << " size=" << size << ": "
-              << ucs_status_string(status)
-              << " memtype=" << mem_buffer::mem_type_name(mem_type);
+        check_lookup(ptr, size, status, mem_type, false);
```

Comment:
can we put ucs_memtype_cache_lookup() call inide check_lookup() to reduce code dup?

Reply:
good catch, done